cql3/statements/batch_statement.cc

```diff
@@ -233,8 +233,8 @@ void batch_statement::verify_batch_size(service::storage_proxy& proxy, const std
             for (auto&& m : mutations) {
                 ks_cf_pairs.insert(m.schema()->ks_name() + "." + m.schema()->cf_name());
             }
-            return format("Batch of prepared statements for {} is of size {:d}, exceeding specified {} threshold of {:d} by {:d}.",
-                    join(", ", ks_cf_pairs), size, type, threshold, size - threshold);
+            return format("Batch of {:d} prepared statements for {} is of size {:d} bytes, exceeding specified {} threshold of {:d} by {:d}.",
+                    mutations.size(), join(", ", ks_cf_pairs), size, type, threshold, size - threshold);
         };
```

Comment:
mutations.size() is not the number of statements, it's the number of tables affected. Two statements that affect one table result in one mutation.

Reply:
Two statements that affect different partitions in one table will result in two mutations.

Reply:
Right. But to statements that affect one partition will create one mutation.

Reply:
Yes. For Cassundra this will be true even if partitions belong to different tables in the same keyspace, but not for us.

Reply:
Sending a v2